src/components/AddressHistory.js

```diff
@@ -0,0 +1,352 @@
+   * transactions {Array} List of transactions on the list
+   * loading {boolean} If is waiting response of data request
+   * firstHash {String} First hash of the list, used in pagination
+   * lastHash {String} Last hash of the list, used in pagination
+   * hasAfter {boolean} If has more elements after the list page
+   * hasBefore {boolean} If has more elements before the list page
+   * queryParams {Object} Object with URL parameters data
+   * numberOfTransactions {Number} Total number of transactions of the list
+   */
+  state = {
+    transactions: [],
+    loading: true,
+    firstHash: null,
+    lastHash: null,
+    hasAfter: false,
+    hasBefore: false,
+    queryParams: null,
+    numberOfTransactions: 0,
+  }
+
+  componentDidMount() {
+    this.setState({ queryParams: this.props.pagination.obtainQueryParams() }, () => {
+      if (this.state.queryParams.token !== null) {
+        this.getData(this.state.queryParams);
+      }
+    });
+
+    WebSocketHandler.on('network', this.handleWebsocket);
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (this.props.address !== prevProps.address) {
+      // This case will be handled by the other component, which will add the token in the URL
+      // and then call this method again later
+      return;
+    }
+
+    const queryParams = this.props.pagination.obtainQueryParams();
+
+    // Do we have new URL params?
+    if (!isEqual(this.state.queryParams, queryParams)) {
+      // Fetch new data, unless query params were cleared and we were already in the most recent page
+      if (queryParams.hash || this.state.hasBefore) {
+        this.getData(queryParams);
+      }
+
+      if (queryParams.token !== this.state.queryParams.token) {
+        // User selected a new token, so we must go to the first page (clear queryParams)
+        this.props.pagination.clearOptionalQueryParams();
+        this.getData(this.props.pagination.obtainQueryParams());
+      }
+    }
+  }
+
+  componentWillUnmount() {
+    WebSocketHandler.removeListener('network', this.handleWebsocket);
+  }
+
+  /**
+   * Called when 'network' ws message arrives
+   * If it's a new tx message update data, in case is necessary
+   *
+   * @param {Object} wsData Data from websocket
+   */
+  handleWebsocket = (wsData) => {
+    if (wsData.type === 'network:new_tx_accepted') {
+      this.updateListWs(wsData);
+    }
+  }
+
+  /**
+   * Update the list with a new tx arrived in a ws message
+   * Checks if the list is on the first page, so it can have a new element
+   * Also validates if this tx is compatible with the list
+   *
+   * @param {Object} tx Data of a newly arrived tx
+   */
+  updateListWs = (tx) => {
+    // We only add new tx/blocks if it's the first page
+    if (!this.state.hasBefore) {
+      if (this.props.shouldUpdate(tx, true)) {
+        let transactions = this.state.transactions;
+        let hasAfter = (this.state.hasAfter || (transactions.length === TX_COUNT && !this.state.hasAfter))
```

Comment:
Final `&& !this.state.hasAfter` seems unnecessary.

Reply:
It does not make any difference but I agree, it's unnecessary. Removed it